bpfman/src/utils.rs

```diff
@@ -301,3 +312,132 @@ pub(crate) fn initialize_bpfman() -> anyhow::Result<()> {
+            BpfmanError::Error(format!(
+                "Failed to open target netns file /var/run/netns/{netns} or /host/netns/{netns}: {e}"
+            ))
+        })?,
+    };
+
+    // Switch to the target network namespace
+    debug!("Switching to the target network namespace: {}", netns);
+    setns(target_netns_file, CloneFlags::CLONE_NEWNET)
+        .map_err(|e| BpfmanError::Error(format!("setns error: {}", e)))?;
+
+    Ok(NetnsGuard {
+        original_ns: bpfman_netns_file,
+    })
+}
+
+macro_rules! xdp_did_string {
+    ($netns:expr, $if_index:expr) => {
+        if let Some(netns) = $netns {
+            format!("{}_{}_{}", XDP_DISPATCHER_PREFIX, netns, $if_index)
+        } else {
+            format!("{}_{}", XDP_DISPATCHER_PREFIX, $if_index)
+        }
+    };
+}
+pub(crate) use xdp_did_string;
+
+macro_rules! tc_did_string {
+    ($netns:expr, $if_index:expr, $direction:expr) => {
+        if let Some(netns) = $netns {
+            format!(
+                "{}_{}_{}_{}",
+                TC_DISPATCHER_PREFIX, netns, $if_index, $direction
+            )
+        } else {
+            format!("{}_{}_{}", TC_DISPATCHER_PREFIX, $if_index, $direction)
+        }
+    };
+}
+pub(crate) use tc_did_string;
+
+macro_rules! xdp_dispatcher_rev_path {
+    ($netns:expr, $if_index:expr, $revision:expr) => {
+        if let Some(netns) = $netns {
+            format!(
+                "{}/dispatcher_{}_{}_{}",
+                RTDIR_FS_XDP, netns, $if_index, $revision
+            )
+        } else {
+            format!("{}/dispatcher_{}_{}", RTDIR_FS_XDP, $if_index, $revision)
+        }
+    };
+}
+pub(crate) use xdp_dispatcher_rev_path;
+
+macro_rules! xdp_dispatcher_link_path {
+    ($netns:expr, $if_index:expr) => {
+        if let Some(netns) = $netns {
+            format!("{}/dispatcher_{}_{}_link", RTDIR_FS_XDP, netns, $if_index)
+        } else {
+            format!("{}/dispatcher_{}_link", RTDIR_FS_XDP, $if_index)
+        }
+    };
+}
+pub(crate) use xdp_dispatcher_link_path;
+
+macro_rules! xdp_dispatcher_link_id_path {
+    ($netns:expr, $if_index:expr, $revision:expr, $id:expr) => {
+        format!(
+            "{}/link_{}",
+            xdp_dispatcher_rev_path!($netns, $if_index, $revision),
+            $id
+        )
+    };
+}
+pub(crate) use xdp_dispatcher_link_id_path;
+
+macro_rules! tc_dispatcher_rev_path {
+    ($direction:expr, $netns:expr, $if_index:expr, $revision:expr) => {{
+        let tc_base = match $direction {
+            Ingress => RTDIR_FS_TC_INGRESS,
+            Egress => RTDIR_FS_TC_EGRESS,
+        };
+        if let Some(netns) = $netns {
+            format!(
+                "{}/dispatcher_{}_{}_{}",
+                tc_base, netns, $if_index, $revision
+            )
+        } else {
+            format!("{}/dispatcher_{}_{}", tc_base, $if_index, $revision)
+        }
+    }};
+}
+pub(crate) use tc_dispatcher_rev_path;
+
+macro_rules! tc_dispatcher_link_id_path {
```

Comment:
These macros should really be functions IMHO.
The benefit of using macros here would be if we were able to do something like:

```
dispatcher_db_key!(tc, direction, netns, if_index); // equivalent to did_string
disptacher_db_key!(xdp, if_index, netns);

dispatcher_link!(tc, direction, netns, if_index, revision); // equivalent to dispatcher_rev_path
dispatcher_link!(tc, direction, netns, if_index, revision, id); // equivalent to dispatcher_link_id_path
```

i.e making use of variadic arguments and/or not having to specify a type for the arguments

Reply:
They are functions now.